backend/services/stats_service.py

```diff
@@ -477,3 +479,87 @@ def update_project_stats(project_id: int):
             Task.task_status == TaskStatus.BADIMAGERY.value
         ).count()
         project.save()
+
+    @staticmethod
+    def get_task_stats(
+        start_date, end_date, org_id, org_name, campaign_id, project_id, country
+    ):
+        """ Creates tasks stats for a period using the TaskStatsDTO """
+
+        query = (
+            db.session.query(
+                TaskHistory.task_id,
+                TaskHistory.project_id,
+                TaskHistory.action_date,
+                Task.task_status,
+            )
+            .distinct(TaskHistory.task_id, TaskHistory.project_id)
+            .join(
+                Task,
+                and_(
+                    Task.id == TaskHistory.task_id,
+                    Task.project_id == TaskHistory.project_id,
+                ),
+            )
+            .filter(
+                Task.task_status.in_(
+                    (
+                        TaskStatus.MAPPED.value,
+                        TaskStatus.VALIDATED.value,
+                        TaskStatus.INVALIDATED.value,
+                        TaskStatus.BADIMAGERY.value,
+                    )
+                )
+            )
+            .order_by(
+                TaskHistory.task_id,
+                TaskHistory.project_id,
+                TaskHistory.id.desc(),
+            )
+        )
+
+        if start_date:
+            query = query.filter(TaskHistory.action_date >= start_date)
+        if end_date:
+            query = query.filter(TaskHistory.action_date <= end_date)
+        if org_id:
+            query = query.join(Project, Project.id == TaskHistory.project_id).filter(
+                Project.organisation_id == org_id
+            )
+        if org_name:
+            organisation = OrganisationService.get_organisation_by_name(org_name)
+            query = query.join(Project, Project.id == TaskHistory.project_id).filter(
+                Project.organisation_id == organisation.id
+            )
+        if campaign_id:
+            query = query.join(
+                campaign_projects,
+                campaign_projects.c.project_id == TaskHistory.project_id,
+            ).filter(campaign_projects.c.campaign_id == campaign_id)
+        if project_id:
+            query = query.filter(TaskHistory.project_id.in_(project_id))
+        if country:
+            # Unnest country column array.
+            sq = Project.query.with_entities(
+                Project.id, func.unnest(Project.country).label("country")
+            ).subquery()
+
+            query = query.filter(sq.c.country.ilike("%{}%".format(country))).filter(
+                TaskHistory.project_id == sq.c.id
+            )
+
+        stats_dto = TaskStatsDTO()
+        stats_dto.mapped = query.filter(
+            Task.task_status.in_((TaskStatus.MAPPED.value, TaskStatus.VALIDATED.value))
```

Comment:
That line should be `Task.task_status == TaskStatus.MAPPED.value`. It's better to return the raw values and sum it on the frontend if needed.